typescript/web/src/connectors/repository/image-processing.ts

```diff
@@ -96,82 +92,39 @@ export const processImage: Repository["imageProcessing"]["processImage"] =
       height,
       mimetype,
       url,
-      id,
       thumbnail20Url,
       thumbnail50Url,
       thumbnail100Url,
       thumbnail200Url,
       thumbnail500Url,
     },
     getImage,
-    putImage,
-    updateImage,
-    user
+    putImage
   ) => {
     const buffer = await getImage(url);
-
     const image = await Jimp.read(buffer as Buffer);
 
-    const result = {
-      width: image.bitmap.width,
-      height: image.bitmap.height,
-      mimetype: image.getMIME(),
-    };
-
-    const thumbnailGenerationInput = {
-      id,
-      image,
-      url,
-      putImage,
-      updateImage,
-      user,
-    };
-
-    if (!thumbnail20Url) {
-      generateThumbnail({
-        size: 20,
-        ...thumbnailGenerationInput,
-      });
-    }
-
-    if (!thumbnail50Url) {
-      generateThumbnail({
-        size: 50,
-        ...thumbnailGenerationInput,
-      });
-    }
-
-    if (!thumbnail100Url) {
+    const generateThumbnailFromSize = (size: ThumbnailSizes) =>
       generateThumbnail({
-        size: 100,
-        ...thumbnailGenerationInput,
+        size,
+        image,
+        url,
+        putImage,
       });
-    }
-
-    if (!thumbnail200Url) {
-      generateThumbnail({
-        size: 200,
-        ...thumbnailGenerationInput,
-      });
-    }
-
-    if (!thumbnail500Url) {
-      generateThumbnail({
-        size: 500,
-        ...thumbnailGenerationInput,
-      });
-    }
 
     return {
-      thumbnail20Url,
-      thumbnail50Url,
-      thumbnail100Url,
-      thumbnail200Url,
-      thumbnail500Url,
       ...validateImageSize({
-        width: width ?? result.width,
-        height: height ?? result.height,
-        mimetype: mimetype ?? result.mimetype,
+        width: width ?? image.bitmap.width,
+        height: height ?? image.bitmap.height,
+        mimetype: mimetype ?? image.getMIME(),
       }),
+      thumbnail20Url: thumbnail20Url ?? (await generateThumbnailFromSize(20)),
+      thumbnail50Url: thumbnail50Url ?? (await generateThumbnailFromSize(50)),
+      thumbnail100Url:
+        thumbnail100Url ?? (await generateThumbnailFromSize(100)),
+      thumbnail200Url:
+        thumbnail200Url ?? (await generateThumbnailFromSize(200)),
+      thumbnail500Url:
+        thumbnail500Url ?? (await generateThumbnailFromSize(500)),
```

Comment:
From what I understand, thumbnails will be generated sequentially here. Isn't it?

Reply:
Good catch, I change it here 0b26d6e972ecce8922c4940abe4fb3a8b5cacbf3